src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_VALUE = 50;
+    private final Random random;
+
+    public FigureSupplier(Random random) {
+        this.random = random;
+    }
+
+    public int randomFigureSupplier(int random) {
+        random = this.random.nextInt(MAX_VALUE);
```

Comment:
`this.` is redundant here, inside the class you can get access to its fields directly